runtime/common/RuntimeMLIR.cpp

```diff
@@ -168,7 +170,7 @@ void registerToQIRTranslation() {
 
         std::unique_ptr<llvm::LLVMContext> llvmContext =
             std::make_unique<llvm::LLVMContext>();
-        llvmContext->setOpaquePointers(false);
+        llvmContext->setOpaquePointers(true);
```

Comment:
Maybe my understanding of this function is wrong, but if we set this to true, then the typed opaque pointers `Array`, `Qubit` etc from the QIR will be replaced with `ptr`. So I had set this to false so that it wouldn't do that, and then we'd be ok with translating to QIR with full opaque type names, which is what is required by the backends that accept QIR at this point.

Reply:
Opened an issue.